### ISSUE TITLE
`tree_sequence.dump()` gives larger file than `tree_sequence.genotype_matrix()`

### ISSUE THREAD

**Issue by agladstein** [tskit-dev/msprime#631]
Hi,

I wanted to do large locus simulations with msprime and print the tree format to read in later. However, I'm finding that `tree_sequence.dump()` is giving a larger file than `tree_sequence.genotype_matrix()`, and I was expecting `tree_sequence.dump()` to be smaller. Is this expected behavior or am I doing something wrong?

Here is the relevant code:

```python
def sim_constant(L):
    tree_sequence = msprime.simulate(sample_size=50,
                                     Ne=10000,
                                     length=L,
                                     recombination_rate=1e-8,
                                    mutation_rate=1.2e-8 / 10,
                                    random_seed=1000)
    return tree_sequence

tree_constant = sim_constant(L)
tree_constant.simplify().dump('sims.trees')
sequence_np_matrix = store_genotype_np(tree_constant)
np.savez_compressed('sims.npz', sequence_np_matrix)
```

I've run it with L = 1Mb, L = 10Mb, L = 50Mb, and L = 100MB.

I'm finding that the .trees files are about 100-150x larger than the .npz files.

If I increase the sample size to 10000, then the .trees files are about 2x larger than the .npz files.

Thanks,
Ariella

**Comment by petrelharp** [comment]
This is possible: the tree sequence file has a lot more information in it
(all the genealogies).  The tradeoff is, roughly, that the tree sequence
file must store each relevant recombination, but given these stores the
mutations almost for free, so if mutation rate is much lower than
recombination rate then the tree sequence file could be larger.  You have
recombination rate about 10x mutation rate, so this is not unexpected.

We talk a bit more about this in
  https://www.biorxiv.org/content/early/2018/06/07/248500
if you are curious.

* peter

**Comment by jeromekelleher** [comment]
It might be worth trying the ``reduce_to_site_topology`` option in simplify, which should reduce the size of your tree sequences a bit in this case. I.e.

```
ts = ts.simplify(reduce_to_site_topology=True)
```

before you save. Because your recombination rate is high compare to mu, this should make a difference. But the compressed genotype matrices will probably still be smaller as your n is quite small.

**Comment by agladstein** [comment]
@jeromekelleher That was a big help!

So, now I am: 
1. simulating without putting down mutations,
2. printing trees with the `reduce_to_site_topology=True` option,
3. later, reading in trees and putting on mutations.

I need the file to be small between steps 2 and 3. With mu = 1.2e-8 / 10, rho = 1e-8, L = 100Mb, sample_size = 50, file size ratio is 
simulate with mutations .trees : simulate with mutations npz : simulate without mutations and reduce_to_site_topology .trees = 1935 : 19 : 1 

I tested that the sequences after I put on mutations with and without `reduce_to_site_topology=True` were the same - so looks like this method works to me. Any issue with it that I haven't thought of?

```python
def sim_constant(L):
    tree_sequence = msprime.simulate(sample_size=50,
                                     Ne=10000,
                                     length=L,
                                     recombination_rate=1e-8,
                                    mutation_rate=1.2e-8 / 10,
                                    random_seed=1000)
    return tree_sequence

tree_constant = sim_constant(L)
tree_constant.simplify(reduce_to_site_topology=True).dump('sims.trees')
```

To use later - 
```
tree_constant = msprime.load('sims.trees')
tree_constant_u = msprime.mutate(tree_constant, rate=u, random_seed=1000)
```

**Comment by agladstein** [comment]
For anyone else interested in this case:
[trees_vs_npz.xlsx](https://github.com/tskit-dev/msprime/files/2543970/trees_vs_npz.xlsx)

**Comment by jeromekelleher** [comment]
No, this won't work I'm afraid. I should have explained what reduce to site topology does. It removes any topology information not needed to represent a given set of sites. If you use it with no sites, it'll get rid of all the topology.

So for your application, you have to keep the mutations on if you want to minimise the tree sequence size.

If space is this critical, you can always zip the trees file. This will make it a good bit smaller.

**Comment by agladstein** [comment]
Ah! That's exactly what I was afraid of. I tried to test for that, but I had a mistake in my test.
Oh well.

Thanks for catching that!
